### ISSUE TITLE
using required() on an object with a nullish() attribute breaks validation and typing

### ISSUE THREAD

**Issue by JabobKrauskopf** [colinhacks/zod#2051]
Please take a look at this code sample:
```typescript
const testCase = z
  .object({
    attributeOne: z.string(),
    attributeTwo: z.string().nullish(),
  })
  .required();

const test = testCase.parse({ attributeOne: "Test" });
```
`testCase.parse({ attributeOne: "Test" })` does not throw an error, even though I would expect it to because of the `required()` function.

These two code samples do however throw the expected error:
```typescript
const testCase = z
  .object({
    attributeOne: z.string(),
    attributeTwo: z.string().optional(),
  })
  .required();

const test = testCase.parse({ attributeOne: "Test" });
```
```typescript
const testCase = z
  .object({
    attributeOne: z.string(),
    attributeTwo: z.string().nullable(),
  })
  .required();

const test = testCase.parse({ attributeOne: "Test" });
```

Using `.optional().nullable()` in place of `.nullish()` also does not produce the expected error.

This not only breaks the validation logic of `.parse()` and `.safeParse()`, the resulting type is also incorrect.
The type of `test` in my first code sample is
```typescript
{
  attributeOne: string;
  attributeTwo: string | null;
}
```
even though `attributeTwo` is `undefined`.

Is this the correct behaviour?

**Comment by JacobWeisenburger** [comment]
Everything is working as expected for me. Try updating to the latest version of zod.

```ts
const schema = z.object( {
    attributeOne: z.string(),
    attributeTwo: z.string().nullish(),
} ).required()

const result = schema.safeParse( { attributeOne: 'Test' } )
!result.success && console.log( result.error.issues )
// [
//     {
//         code: 'invalid_type',
//         expected: 'string',
//         received: 'undefined',
//         path: [ 'attributeTwo' ],
//         message: 'Required'
//     }
// ]
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Comment by JabobKrauskopf** [comment]
Thanks for the help!

That fixed the problem! I had version `3.20.2` installed - now using `3.20.6` and everything works as expected.

Sorry for the non-issue 😅

**Comment by JacobWeisenburger** [comment]
no problem. I had the same problem yesterday, that I was using an older version and updated and then everything was working fine. 😆
